angr/analyses/cfg/cfg_fast.py

```diff
@@ -3591,7 +3612,7 @@ def _generate_cfgnode(self, cfg_job, current_function_addr):
 
                 return addr, current_function_addr, cfg_node, irsb
 
-            is_arm_arch = self.project.arch.name in ('ARMHF', 'ARMEL')
+            is_arm_arch = True if self.project.arch.name in ('ARMHF', 'ARMEL', 'ARMCortexM') else False
```

Comment:
Redundant. Should be `is_arm_arch = self.project.arch.name in {'ARMHF', 'ARMEL', 'ARMCortexM'}`.

Reply:
This is another general concern I have, we do checks like this _everywhere_ and there is no consistent way to say "if i'm on x arch or group of arches or arch family" other than this gross hardcoding thing or a subclass check (which I guess we avoid for speed reasons?)
is_arm_arch() is fine, but it would be nice to have a generic approach
Or we just add is_mips_arch, is_ppc_arch, and is_intel_arch and be done with it i guess?
How are you handling this for the decompiler and its probably-necessary arch-specific fixups?

Reply:
Created an interim solution, as we discussed on slack.